### PR TITLE
WCS clip and ship

### PR DESCRIPTION
**Pull request by tephenavies**
Add WCS export capability for WMS items that have a `linkedWcsUrl` and `linkedWcsCoverage`. Includes a fix for rectangle parameter selection in Leaflet.

### CODE REVIEW

**Comment by tephenavies** [comment]
See #2937

File: lib/Models/WebMapServiceCatalogItem.js [hunk 1]
```diff
@@ -70,6 +71,18 @@ var WebMapServiceCatalogItem = function(terria) {
      */
     this.layers = '';
 
+    /**
+     * Gets or sets the URL of a WCS that enables clip-and-ship for this WMS item. This property is observable.
+     * @type {String}
+     */
+    this.linkedWcsURL = undefined;
```
Comment (kring) [inline lib/Models/WebMapServiceCatalogItem.js]: Nitpick: should be `Url` not `URL`.

File: lib/ReactViews/Workbench/Controls/ViewingControls.jsx [hunk 2]
```diff
@@ -107,6 +112,10 @@ const ViewingControls = createReactClass({
                 <If condition={canSplit}>
                     <li className={classNames(Styles.split, classList)}><button type='button' onClick={this.splitItem} title="Duplicate and show splitter" className={Styles.btn}>Split</button></li>
                 </If>
+                <If condition={defined(item.linkedWcsURL)}>
+                    {/* WIP: Fix styles */}
```
Comment (kring) [inline lib/ReactViews/Workbench/Controls/ViewingControls.jsx]: Do you still need this WIP comment?

File: lib/Models/callWebCoverageService.js [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+var sprintf = require('terriajs-cesium/Source/ThirdParty/sprintf');
+
+var MapInteractionMode = require('./MapInteractionMode');
+var proxyCatalogItemUrl = require('./proxyCatalogItemUrl');
+var ResultPendingCatalogItem = require('./ResultPendingCatalogItem');
+var TerriaError = require('../Core/TerriaError');
+
+
+const callWebCoverageService = function(wmsCatalogItem) {
+//     this.terria = terria;
```
Comment (kring) [inline lib/Models/callWebCoverageService.js]: Mind cleaning up this commented-out code?

**Comment by kring** [comment]
Doesn't work in IE11:
image

image

When I was working on the map/assets export feature awhile back, I used the `file-saver` library, which seemed to work pretty well:
https://github.com/TerriaJS/terriajs/blob/export/lib/ReactViews/Map/Panels/SharePanel/SharePanel.jsx#L398

**Comment by kring** [comment]
The only other thing I might suggest is that you add some comments around the new properties on WMSCatalogItem saying it's an experimental feature and subject to change. Cause it is, right?

**Comment by tephenavies** [comment]
I have fixed most of these problems. IE11 looks like it's still a little troublesome (3D rendering failed as soon as I started a selection for exporting, but it worked when the page was refreshed and retried in 2D).

Styles are still WIP. Not sure how to implement them without adding another dimension of complexity to this: https://github.com/TerriaJS/terriajs/blob/9028f69c990af8f33f1cf47a723bcc3d6cb6a05e/lib/ReactViews/Workbench/Controls/viewing-controls.scss#L43-L108

**Pull request merged**